plugins/modules/mso_schema_site_external_epg.py

```diff
@@ -160,7 +214,17 @@ def main():
             ops.append(dict(op="remove", path=op_path))
 
     elif state == "present":
-        l3out_dn = "uni/tn-{0}/out-{1}".format(tenant_name, l3out)
+        # Get external EPGs type from template level
+        external_epgs = [e.get("name") for e in schema_obj.get("templates")[template_idx]["externalEpgs"]]
+        if external_epg is not None and external_epg in external_epgs:
+            external_epg_idx = external_epgs.index(external_epg)
+            ext_epg_type = schema_obj.get("templates")[template_idx]["externalEpgs"][external_epg_idx].get("extEpgType")
+            if ext_epg_type != "cloud":
+                if l3out is not None:
+                    l3out_dn = "uni/tn-{0}/out-{1}".format(tenant_name, l3out)
+                else:
+                    mso.fail_json(msg="L3Out cannot be empty when template external EPG type is 'on-premise'.")
```

Comment:
Should we use static 'on-premise' or also considered using the ext_epg_type with formatted string?

Reply:
static on-premise.